File: 23.tell-dont-ask/src/typescript/tests/doubles/testShipmentService.ts
import { Order } from '../../domain/order';
import { ShipmentService } from '../../service/shipmentService';

export class TestShipmentService implements ShipmentService {
  private shippedOrder: Order | undefined;

  ship(order: Order): void {
    this.shippedOrder = order;
  }

  getShippedOrder(): Order | undefined {
    return this.shippedOrder;
  }
}

File: 23.tell-dont-ask/src/java/src/test/java/com/telldontask/doubles/TestShipmentService.java
package com.telldontask.doubles;

import com.telldontask.domain.Order;
import com.telldontask.service.ShipmentService;

public class TestShipmentService implements ShipmentService {
    private Order shippedOrder;

    @Override
    public void ship(Order order) {
        this.shippedOrder = order;
    }

    public Order getShippedOrder() {
        return shippedOrder;
    }
}
